library/std/src/sys/hermit/os.rs

```diff
@@ -29,7 +29,7 @@ pub fn chdir(_: &path::Path) -> io::Result<()> {
     unsupported()
 }
 
-pub struct SplitPaths<'a>(&'a Void);
+pub struct SplitPaths<'a>((!, &'a ()));
```

Comment:
I changed the type from `&'a !` to `(!, &'a ())`, because the semantics of references to uninhabited types changed: where previously `&!` was uninhabited, now it is not.

Reply:
Maybe a combination of `!` and `PhantomData<&'a ()>` expresses the intention better.